lib/iris/fileformats/netcdf/_thread_safe_nc.py

```diff
@@ -35,22 +35,42 @@ class _ThreadSafeWrapper(ABC):
     the C-layer.
     """
 
+    # Note: this is only used to create a "contained" from passed args.
     CONTAINED_CLASS = NotImplemented
+    # Note: this defines how we identify/check that a contained is of the expected type
+    # (in a duck-type way).
+    _DUCKTYPE_CHECK_PROPERTIES: typing.List[str] = [NotImplemented]
 
     # Allows easy type checking, avoiding difficulties with isinstance and mocking.
     THREAD_SAFE_FLAG = True
 
+    @classmethod
+    def is_contained_type(cls, instance):
+        return all(
+            hasattr(instance, attr) for attr in cls._DUCKTYPE_CHECK_PROPERTIES
+        )
+
     @classmethod
     def _from_existing(cls, instance):
```

Comment:
Do you think we should make this public since it's being used elsewhere now?

Alternatively: if we want to distinguish our dedicated class methods from the original methods of the NetCDF4 classes, then we should make `is_contained_type()` private.

I'm happy either way, but the same should definitely apply to both methods.